### ISSUE TITLE
Chain Coordinates instead of TxnID

### ISSUE THREAD

**Issue by ThinkingJoules** [ThinkingJoules/distributed-web-data#2]
If we can avoid using the hash of a txn and instead use the serialized nature of blockchains that we are paying a price for, can we safely use one or two uints to address a txn or state within a chain? 

The idea is that the chain is more or less an ever-growing array. The index is the block, and, depending on configuration on said chain, the second uint is the txn within the block. I could imagine really expensive, try-not-to-grow the-state-chains, having block=txn to allowing the dropping of the 2nd dimension.

If we can use uints, we now have something that is much shorter than a hash and could use it directly as labels to massively cut down on identifier costs, in both storage and bandwidth. If there might be trillions of identifiers, lowering them from 32 bytes to something like an average length of 4-5 bytes is considerable at internet scale.

The only failure mode I can think of would be a Chain re-org. I think with PoS style consensus irreversibility comes much faster and permanently, than with something like Bitcoin. However, if an attack occurs, could irreversibility be upheld? According to the AVAX [whitepaper](https://assets.website-files.com/5d80307810123f5ffbb34d6e/6009805681b416f34dcae012_Avalanche%20Consensus%20Whitepaper.pdf) in section 4.2 "Liveness" it states that (emphasis mine): 
> We assume that the observed adversarial presence 0 ≤ f 0 ≤
n(k −α−ψ)/k ≤ f , where we refer to ψ as the buffer zone.
The biggerψ, the quicker the ability of the decision mechanism
to finalize a value. If, of course, ψ approaches zero or becomes
negative, then we violate the upper bound of adversarial
tolerance for the parameterized system, and thus the adversary
can, with high probability, stall termination by simply choosing
to not respond, although the ***safety guarantees may still hold.***

The "safety guarantee" is the reversibility of previous state before an adversarial attack. So basically an attack would stop the chain from progressing, like a DDoS attack, but it doesn't appear that it can re-org blocks like in bitcoin. Have to investigate when the "may hold" is "will hold".

# Upshot
So basically, if irreversibility holds, and there is no confusion on *which* chain we are using the coordinates, then this has the properties of a hash (one->one mapping, globally unique) but at a massively reduced data footprint.

The added benefit is that the most common symbols will be created first thus also giving minimum system-level data 'compression'.
